redbox-core/redbox/models/chain.py

```diff
@@ -0,0 +1,280 @@
+from datetime import UTC, datetime
+from enum import StrEnum
+from functools import reduce
+from typing import (
+    Annotated,
+    Literal,
+    TypedDict,
+)
+from uuid import UUID, uuid4
+
+from langchain_core.documents import Document
+from langchain_core.messages import AnyMessage
+from langgraph.graph.message import add_messages
+from langgraph.managed.is_last_step import RemainingStepsManager
+from pydantic import BaseModel, Field
+
+from redbox.models import prompts
+from redbox.models.settings import ChatLLMBackend
+
+
+class ChainChatMessage(TypedDict):
+    role: Literal["user", "ai", "system"]
+    text: str
+
+
+class AISettings(BaseModel):
+    """Prompts and other AI settings"""
+
+    # LLM settings
+    context_window_size: int = 128_000
+    llm_max_tokens: int = 1024
+
+    # Prompts and LangGraph settings
+    max_document_tokens: int = 1_000_000
+    self_route_enabled: bool = False
+    map_max_concurrency: int = 128
+    stuff_chunk_context_ratio: float = 0.75
+    recursion_limit: int = 50
+
+    # Common Prompt Fragments
+
+    system_info_prompt: str = prompts.SYSTEM_INFO
+    persona_info_prompt: str = prompts.PERSONA_INFO
+    caller_info_prompt: str = prompts.CALLER_INFO
+
+    # Task Prompt Fragments
+
+    chat_system_prompt: str = prompts.CHAT_SYSTEM_PROMPT
```

Comment:
Do we just need one system prompt now?

Reply:
maybe two (one for chat-with-docs, the other for without?)

Reply:
Yes, good point!